### PR TITLE
Add yvUSDC as collateral

### PR DESCRIPTION
**Pull request by devtooligan**
# Proposal
Enable Yearn USDC Vaults (yvUSDC) as a collateral for USDC loans

# Background
The vault token yvUSDC (api 0.4.3) has over $154m USD market cap and total value locked in all Yearn vaults is over $3.8 billion. Adding yvUSDC would allow us to gauge customer interest and inform our decision on adding other Yearn vault types. Furthermore, the yvUSDC price is based on USDC and we are initially only allow borrowing against it in USDC, so that makes it a lower risk and an acceptable entry point into Yearn vault collateral.

To enable yvUSDC as a collateral we will be using the [pricePerShare](https://github.com/yearn/yearn-vaults/blob/main/contracts/Vault.vy) function on the [yvUSDC token contract](https://etherscan.io/token/0xa354F35829Ae975e850e23e9615b11Da1B3dC4DE#readContract).  This function calculates the current share value in real time as: ` total vault assets / total token supply`.

# Details

The [addYVUSDC-1.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-1.ts), [addYVUSDC-2.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-2.ts)  and [addYVUSDC-3.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-3.ts) scripts  will be used, with [this input](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC.mainnet.config.ts):

```

export const assets: Map<string, string> = new Map([
  [USDC,   '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'],
  [YVUSDC, '0xa354F35829Ae975e850e23e9615b11Da1B3dC4DE'],
])

// underlying, yvToken, address
export const yearnSources: Array<[string, string, string]> = [
  [USDC, YVUSDC,  assets.get(YVUSDC) as string]
]

// Assets for which we will have a Join
export const assetsToAdd: Array<[string, string]> = [
  [YVUSDC, assets.get(YVUSDC) as string],
]

// Input data: baseId, ilkId, oracle name, ratio (1000000 == 100%), inv(ratio), line, dust, dec
export const yearnLimits: Array<[string, string, string, number, number, number, number, number]> = [
  [USDC, YVUSDC, YEARN, 1250000, 800000, 1000000,  5000,    6],
]

// Input data: seriesId, [ilkIds]
export const seriesIlks: Array<[string, string[]]> = [
  [FYUSDC2112, [YVUSDC]],
  [FYUSDC2203, [YVUSDC]],
]
```
Steps:
 1.  Deploy YearnOracle and give ROOT access to the Timelock.
 2.  Configure the permissions for the Yearn Oracle
 3.  Add the USDC/YVUSDC yvToken as the USDC/YVUSDC source in the Yearn Oracle
 7.  Add YVUSDC as an asset to the Yield Protocol using the Wand
 8.  Permission the YVUSDCJoin
 9.  Make YVUSDC into an ilk
 10. Approve YVUSDC as collateral for USDC series

# Testing
The change has been tested on a mainnet fork by posting yvUSDC to all series, borrowing, repaying and withdrawing.  Here are the results of the [tests](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC.test.ts):

```
$ npx hardhat run --network localhost scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC.test.ts
No need to generate any newer typings.
ChainId: 1
1002007137043 YVUSDC available
series: 0x303230340000
vault: 0xfc9e74c26b167d7826d9c2a8
posted and borrowed
repaid and withdrawn
series: 0x303230350000
vault: 0xea90d3561c60bf79271a0eac
posted and borrowed
repaid and withdrawn
```

### CODE REVIEW

File: YPP-0023.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+]
+
+// Assets for which we will have a Join
+export const assetsToAdd: Array<[string, string]> = [
+  [YVUSDC, assets.get(YVUSDC) as string],
+]
+
+// Input data: baseId, ilkId, oracle name, ratio (1000000 == 100%), inv(ratio), line, dust, dec
+export const yearnLimits: Array<[string, string, string, number, number, number, number, number]> = [
+  [USDC, YVUSDC, YEARN, 1400000, 714000, 500000,  5000,    6],
```
Comment (aniemerg) [inline YPP-0023.md:39]: We are doing an 80% loan-to-value (125% collateralization ratio).

File: YPP-0023.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import { USDC, YVUSDC } from '../../../../shared/constants'
+import { YEARN } from '../../../../shared/constants'
+import { FYUSDC2112, FYUSDC2203 } from '../../../../shared/constants'
+
+export const developer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+export const deployer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+
+export const assets: Map<string, string> = new Map([
+  [USDC,   '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'],
+  [YVUSDC, '0x5f18c75abdae578b483e5f43f12a39cf75b973a9'],
```
Comment (aniemerg) [inline YPP-0023.md:24]: The YVUSDC address seems correct: it points to "yvUSDC 0.3.0", but i'm not sure how to confirm that is the right address.

**Review by aniemerg — CHANGES_REQUESTED** [review]
Please update the loan-to-value.

File: YPP-0023.md (hunk 1 above)
Comment (aniemerg) [reply YPP-0023.md:39]: Also, I think we can start with a million dollar borrowing limit at first

Comment (devtooligan) [reply YPP-0023.md]: 👍🏼

File: YPP-0023.md (hunk 2 above)
Comment (devtooligan) [reply YPP-0023.md]: continued this discussion in slack.  that 0x5f18 is an older version of yvUSDC which uses the 0.3.0 api

we've decided to go with the 0xa354F35829Ae975e850e23e9615b11Da1B3dC4DE (yvUSDC api 0.4.3) per Allan's suggestion

**Review by alcueca — APPROVED** [review]
LTV updated, including initialOffer. Shame on the yvUSDC split.

**Comment by devtooligan** [comment]
0x55c5daef1625fdd959f0c1a45515d730c4537d310f025436610e65931978a54d
Screen Shot 2021-12-16 at 6 43 08 AM

**Comment by devtooligan** [comment]
The deployer role in the above proposed txn was incorrect.  Corrected and recreated proposal:

`0x5b11b10ad33b1d285a830a3be4e7abd71e90e0775c84d7c8e2fad95b4a60d005`

Screen Shot 2021-12-16 at 6 58 24 AM

**Comment by devtooligan** [comment]
Executed 1/3
https://etherscan.io/tx/0x231dcb1aa83dde789c55bd212485859338c63e41bf84508d3ca633b7545e6e76

**Comment by devtooligan** [comment]
Proposed step 2/3
`0x314e21e6d1452c9dc64fe32dae5d3eeade98c9a1b5d8a1a2af82d1cf9e64cfc6`

Screen Shot 2021-12-16 at 7 44 27 AM

**Comment by devtooligan** [comment]
executed 2/3
https://etherscan.io/tx/0x7ef4d0da5eaf301da6f7b4d1a7ea83da9c4f57287e0e5a91e9df54eb754f3bc2

**Comment by devtooligan** [comment]
proposed 3/3
0x5550018e0410d3289cdfec54c03a6d8c1def1f08455d701fed3a04d05583194d

**Comment by devtooligan** [comment]
executed 3/3
https://etherscan.io/tx/0x16da41d8eba21060295102c104d1b610f8d78bd61ea7582a2f6c6d97e845e295

**Pull request merged**